training/controllers/mail.py

```diff
@@ -0,0 +1,134 @@
+from flask import Blueprint, request, render_template, url_for, redirect
+from flask_mail import Message
+from itsdangerous import URLSafeTimedSerializer, SignatureExpired
+import os
+
+from training.controllers.forms import new_password_form, recover_form, send_email_form
+from training.extensions import mail, bcrypt
+from training.models.users import User
+from training.extensions import db
+
+""" 
+    Problema original, solucionado. Sin embargo, estaria bueno discutir la solucion. Se relaciona con las background tasks para enviar mails por redis, ya que como 
+    me habias dicho Joaco, tenia que buscar una amanera de pasarle el app_context a redis. 
+
+    Mi problema original era el siguiente:
+    Este modulo es llamado cuando se esta a la mitad e create_app.
+    Recordemos que app, es creado como variable global al final de correrse create_app y aqui estamos en la mitad del flujo.
+    Por lo que cuando importamos "from training.background_tasks.tasks import send_email_async" este importa app desde app.py, pero aun no esta creada ya que
+    nos encontrabamos a mitad del flujo de create_app, y recordemos que app se crea cuando termina create_app. Ergo, ERROR
+    
+    Solucion:
+    Crear la app en 2 pasos. Se hace app=create_app(), y en create_app se inicializa, importa y crea todo salvo este modulo.
+    Luego, se llama a una funcion modifications(), que importa solo este modulo y se agrega el blueprint a la app ya existente.
+    
+    Entonces al importarse este modulo, se comienza a correr, y en determinado momento se importa app desde aca. 
+    Pero ya existe una app, por lo que no hay problema en ir a buscarla a app.py. Y se le puede dar el app_context a redis.
+"""
+
+
+s = URLSafeTimedSerializer(os.environ.get('SECRET_KEY'))
```

Comment:
can you use more descriptive variables?

Reply:
Descriptive variables make the code more readable and understanable for other developers and even for you in the future

Reply:
Sure, i will change it